android/app/src/main/java/app/organicmaps/settings/SettingsPrefsFragment.java

```diff
@@ -258,7 +258,8 @@ private void initSearchPrivacyPrefsCallbacks()
     ((TwoStatePreference) pref).setChecked(isHistoryEnabled);
     pref.setOnPreferenceChangeListener((preference, newValue) -> {
       boolean newVal = (Boolean) newValue;
-      if (newVal != isHistoryEnabled)
+      boolean currVal = Config.isSearchHistoryEnabled();
+      if (newVal != currVal)
```

Comment:
Will this work?
```
    pref.setOnPreferenceChangeListener((preference, Boolean newValue) -> {
      if (newValue != Config.isSearchHistoryEnabled())
```

Reply:
Yes ofcourse but i did that to keep it in proper format

Reply:
What is "proper format"?

Reply:
Like declaring a variable first storing value into that and then using it for further :)

Reply:
So you likely meant "a more complex code to read", right?

Reply:
Yes, but easy to understand.
But since you suggested so i have changed the code accordingly : )

Reply:
My suggestion was one line shorter )

Reply:
Yes i did that :)

Reply:
> My suggestion was one line shorter )

I missed that another line that time. Since it was pre written code so i pre assumed that i will not be asked to delete the pre written codes : )